private/bufpkg/bufimage/bufimage.go

```diff
@@ -129,8 +129,14 @@ func NewMultiImage(images ...Image) (Image, error) {
 // be treated as non-imports in the returned Image. The first non-import
 // version of a file will be used in the result.
 //
+// Note that the files within each image are expected to be de-duplicated
+// before being passed into this function. In the case that two images
+// consider the same file as a non-import, the first will be used. We purposefully
+// do not validate this condition here as a result of the --path filter, and
+// instead validate it as part of the ParserAccessorHandler when each Image
+// is constructed.
+//
 // Reorders the ImageFiles to be in DAG order.
-// Duplicates can exist across the Images, but only if duplicates are non-imports.
```

Comment:
add a comment about duplicates being allowed to exist, also say that formerly non-imports couldnt be duplicated, but why this was removed